File: tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/action/DataGenOnly.scala
package io.glutenproject.integration.tpc.action

import io.glutenproject.integration.tpc.TpcSuite

import java.io.File

case class DataGenOnly(scale: Double, genPartitionedData: Boolean) extends Action {
  override def execute(tpcSuite: TpcSuite): Boolean = {
    tpcSuite.sessionSwitcher.useSession("baseline", "Data Gen")
    val dataGen = tpcSuite.createDataGen(scale, genPartitionedData)
    dataGen.gen()
    true
  }
}

File: tools/gluten-it/common/src/main/java/io/glutenproject/integration/tpc/command/DataGenOnly.java
package io.glutenproject.integration.tpc.command;

import io.glutenproject.integration.tpc.TpcMixin;
import picocli.CommandLine;

import java.util.concurrent.Callable;

@CommandLine.Command(name = "data-gen-only", mixinStandardHelpOptions = true,
    showDefaultValues = true,
    description = "Generate data only.")
public class DataGenOnly implements Callable<Integer> {
  @CommandLine.Mixin
  private TpcMixin mixin;

  @CommandLine.Mixin
  private DataGenMixin dataGenMixin;

  @Override
  public Integer call() throws Exception {
    return mixin.runActions(dataGenMixin.makeActions());
  }
}
